nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/EmbeddedQuestDbRolloverHandler.java

```diff
@@ -91,7 +95,8 @@ private void rolloverTable(final CharSequence tableName) {
 
     private void deletePartition(final CharSequence tableName, final String partition) {
         try (final SqlCompiler compiler = dbContext.getCompiler()) {
-            compiler.compile(String.format(DELETION_QUERY, new Object[]{tableName, partition}), dbContext.getSqlExecutionContext());
+            final CompiledQuery compile = compiler.compile(String.format(DELETION_QUERY, new Object[]{tableName, partition}), dbContext.getSqlExecutionContext());
+            compile.execute(new SCSequence(new TimeoutBlockingWaitStrategy(5, TimeUnit.SECONDS)));
```

Comment:
Prior to this change, was the delete query not being executed?

Reply:
Yes it was. In this particular case, 4.x did not need a call on the execute. At the time `EmbeddedQuestDbRolloverHandlerTest` was not ignored and they showed that the deletion had affect on the database. Also manual testing had the same results.